packages/vector_graphics_compiler/lib/src/debug_format.dart

```diff
@@ -0,0 +1,184 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:vector_graphics_codec/vector_graphics_codec.dart';
+
+/// Write an unstable but human readable form of the vector graphics binary
+/// package intended to be used for debugging and development.
+Uint8List dumpToDebugFormat(Uint8List bytes) {
+  const VectorGraphicsCodec codec = VectorGraphicsCodec();
+  final _DebugVectorGraphicsListener listener = _DebugVectorGraphicsListener();
+  final DecodeResponse response =
+      codec.decode(bytes.buffer.asByteData(), listener);
+  if (!response.complete) {
+    codec.decode(bytes.buffer.asByteData(), listener, response: response);
+  }
+  return utf8.encode(listener.buffer.toString()) as Uint8List;
+}
+
+String _intToColor(int value) {
+  return 'Color(0x${(value & 0xFFFFFFFF).toRadixString(16).padLeft(8, '0')})';
+}
+
+class _DebugVectorGraphicsListener extends VectorGraphicsCodecListener {
+  final StringBuffer buffer = StringBuffer();
+
+  @override
+  void onClipPath(int pathId) {
+    buffer.writeln('DrawClip: id:$pathId');
+  }
+
+  @override
+  void onDrawImage(int imageId, double x, double y, double width, double height,
+      Float64List? transform) {
+    buffer.writeln(
+        'DrawImage: id:$imageId (Rect.fromLTWH($x, $y, $width, $height), transform: $transform)');
+  }
+
+  @override
+  void onDrawPath(int pathId, int? paintId, int? patternId) {
+    buffer.writeln('DrawPath: id:$pathId ($paintId, $patternId)');
```

Comment:
nit: omit pattern id if null, and/or include a label for it?

Reply:
Done